service_crategen/src/commands/generate/codegen/mod.rs

```diff
@@ -224,19 +234,21 @@ pub fn get_rust_type(service: &Service,
     }
                      }
 
-fn has_streaming_member(name: &str, shape: &Shape) -> bool {
-    shape.shape_type == ShapeType::Structure &&
-    shape.members.is_some() &&
-    shape.members.as_ref()
-                 .unwrap()
-                 .iter()
-                 .any(|(_, member)| member.shape == name && member.streaming())
+fn streaming_members<'a>(shape: &'a Shape) -> Box<Iterator<Item = &'a Member> + 'a> {
+    if shape.members.is_some() {
```

Comment:
I just realized I lost the `ShapeType::Structure` check here. Is it necessary? The code runs without it...